### PR TITLE
Generate API-reference also

### PR DESCRIPTION
**Pull request by arve0**
Docstrings are already nice formatted, lets make use of it!

### CODE REVIEW

**Comment by arve0** [comment]
Generated a demo build: http://dask-doc-demo.readthedocs.org/en/apidocs/index.html

**Comment by arve0** [comment]
The readthedocs build fails on subpackages, but works locally, so thats probably just the setup over at readthedocs.

**Comment by shoyer** [comment]
In my experience, you get much more useable API docs if you do some manual organization and explicitly call out each function/class you want to document. Otherwise you end up with too much nesting.

**Comment by arve0** [comment]
I agree. I do not have enough knowledge about the dask API to manually organize it, but this gives a good starting point. Also, scaffolding the API-docs once in a while is a good idea, its easy to miss putting a new submodule in the docs.

**Comment by mrocklin** [comment]
Agree with both.  I'll see if I can find some time this or next week to curate this a bit.

**Comment by arve0** [comment]
Removed the unnecessary `sphinxcontrib-napoleon` which is included in `sphinx>=1.3`.

Also removed one level of nesting. API-reference now goes directly to the dask-package. There is at most 2 steps of nesting now, API-reference->subpackage.

File: dask/utils.py [hunk 1]
```diff
@@ -139,15 +139,14 @@ def textblock(file, start, stop, compression=None):
...
+    >>> with open('myfile.txt', 'w') as f:
+    ...     f.write('123\\n456\\n789\\nabc')
+    >>> f = open('myfile.txt')
 
     In the example below, 1 and 10 don't line up with endlines
 
-    >> textblock(f, 1, 10)
-    '456\n789\n'
+    >>> textblock(f, 1, 10)
+    '456\\n789\\n'
```
Comment (mrocklin) [inline dask/utils.py]: This was actually intentional.  Some doctesters don't parse results with endlines well.

**Comment by mrocklin** [comment]
I'm going to close this in favor of solutions like #236 .  Thanks for spurring this on!

**Pull request closed without merging**